Objects/bytearrayobject.c

```diff
@@ -127,40 +127,35 @@ PyObject *
 PyByteArray_FromStringAndSize(const char *bytes, Py_ssize_t size)
 {
     PyByteArrayObject *new;
-    Py_ssize_t alloc;
 
     if (size < 0) {
         PyErr_SetString(PyExc_SystemError,
             "Negative size passed to PyByteArray_FromStringAndSize");
         return NULL;
     }
 
-    /* Prevent buffer overflow when setting alloc to size+1. */
+    /* Prevent buffer overflow when setting alloc to size. */
     if (size == PY_SSIZE_T_MAX) {
```

Comment:
You can remove this test, PyBytes_FromStringAndSize() has a stricter test on size maximum value:

```c
    if ((size_t)size > (size_t)PY_SSIZE_T_MAX - PyBytesObject_SIZE) {
        PyErr_SetString(PyExc_OverflowError,
                        "byte string is too large");
        return NULL;
    }
```